### ISSUE TITLE
Inheritance issues

### ISSUE THREAD

**Issue by d-dantte** [reinforced/Reinforced.Typings#31]
I find that using the Fluent api, my classes being exported to interfaces do not maintain their inheritance chain. I have also not yet found any way to configure it to work. Is it possible i am missing something, or is the feature not implemented?

**Comment by pavel-b-novikov** [comment]
Have you also exported base classes/interfaces?

**Comment by d-dantte** [comment]
Yes i did. I'll just go ahead and drop my entire code here for you to see.
Thanks for the speedy response.


```
public static class ReinforcedTypings
{
    public static void Configure(ConfigurationBuilder config)
    {
        var polluxBase = typeof(PolluxEntity<>);

        var types = new[] { polluxBase }.Concat(
        polluxBase
            .Assembly
            .GetTypes()
            .Where(_t => _t.IsClass)
            .Where(_t => _t != polluxBase)
            .Where(_t => _t.HasGenericAncestor(polluxBase))); //filters classes who's generic ancestor has the argument as it's generic definition
            
        config.ExportAsInterfaces(types, _icb =>
        {
            _icb.OverrideNamespace("Pollux.Models")
                .WithPublicProperties();
        });
    }
}
```

Examples of classes being exported are below

```
public abstract class PolluxEntity<Key>
{
    public Key EntityId { get; set; }

    public DateTime CreatedOn { get; set; }

    public DateTime? ModifiedOn { get; set; }


    public PolluxEntity()
    {
        this.CreatedOn = DateTime.Now;
    }
}


public class ContactData: PolluxEntity<long>
{
    public string Phone  { get; set; }
    public string AlternatePhone  { get; set; }
    public bool PhoneConfirmed { get; set; }

    public string Email  { get; set; }
    public string AlternateEmail  { get; set; }
    public bool EmailConfirmed  { get; set; }
    
    public virtual string OwnerId  { get; set; }

    public ContactData()
    {
    }
}
```

...and many other such model/entity classes.

But the final result looks like this:
```

export interface IPolluxEntity<Key>
{
    EntityId: Key;
    CreatedOn: any;
    ModifiedOn?: any;
}
export interface IContactData
{
    Phone: string;
    AlternatePhone: string;
    PhoneConfirmed: boolean;
    Email: string;
    AlternateEmail: string;
    EmailConfirmed: boolean;
    OwnerId: string;
}
```

I'm wondering what i'm missing...

Many thanks again in advance.

**Comment by pavel-b-novikov** [comment]
Ok, I will check little bit later. Now I dont have enough time

**Comment by pavel-b-novikov** [comment]
Ok, your issue is fixed in 1.30 branch. Please check out this branch and run your test.

**Comment by d-dantte** [comment]
Will do.
Cheers.

**Comment by pavel-b-novikov** [comment]
If you come up with some other cases to test - feel free to add your tests and contribute. 
You will speed up release of 1.3 and also improve stability

**Comment by pavel-b-novikov** [comment]
Fixed in 1.3.0
